src/Utils/obiDialogParser.ts

```diff
@@ -0,0 +1,331 @@
+        for (const key of keys) {
+            if (!key.startsWith(">")) {
+                const inputs = key.split('- ').map(i => i.trim())
+                luMap.set(inputs[0], inputs.slice(1))
+            }
+        }
+        return luMap
+    }
+    
+    private addToLGMap(text: string, lgMap: Map<string, OBIUtils.LGItem>): any {
+        const items = text.split('# ')
+        for (const item of items) {
+            const key = item.substring(0, item.indexOf("-")).trim()
+            const body = item.substring(item.indexOf("```") + 3, item.lastIndexOf("[")).trim()
+            const suggestionList = item.substring(item.lastIndexOf("[Suggestions=") + 13, item.lastIndexOf("]"))
+            const suggestions = suggestionList.length > 0 ? suggestionList.split('|') : []
+            const output = {
+                text: body,
+                suggestions
+            }
+            lgMap.set(key, output)
+        }
+        return lgMap
+    }
+    
+    private async getTrainDialogsfromOBIDialog(obiDialog: OBIDialog): Promise<CLM.TrainDialog[]> {
+    
+        let trainDialogs: CLM.TrainDialog[] = []
+        if (obiDialog.rules) {
+            for (const rule of obiDialog.rules) {
+                if (rule.$type === "Microsoft.IntentRule") {
+    
+                    const textVariations = this.getTextVariations(rule.intent!)
+                    const extractorStep: CLM.TrainExtractorStep = {
+                        textVariations: textVariations
+                    }
+    
+                    if (rule.steps) {
+                        for (const step of rule.steps) {
+                            if (typeof step === "string") {
+                                throw new Error("Unexpected string step")
+                            }
+                            else {
+                                if (step.$type === "Microsoft.BeginDialog" && typeof step.dialog === "string") {
+                                    
+                                    const subDialog =  this.composerDialog.dialogs.find(d => d.$id === step.dialog)
+                                    if (!subDialog) {
+                                        throw new Error(`Dialog name ${step.dialog} undefined`)
+                                    }
+    
+                                    const childDialogs = await this.getTrainDialogsfromOBIDialog(subDialog)
+                                    
+                                    // Add extractor step to all the children
+                                    childDialogs.forEach(td => {
+                                        td.rounds[0].extractorStep = extractorStep
+                                    })
+    
+                                    // Add children to train dialog list
+                                    trainDialogs = [...trainDialogs, ...childDialogs]
+                                }
+                                else if (step.$type === "Microsoft.SendActivity") {
+    
```

Comment:
Something supposed to happen here? This is weird to be empty. Would at least put a comment saying it's intentionally skipping this `$type`, will be added later, or if expected not to happen do some sort of `throw new Error('Not Implemented')`